clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkThread.java

```diff
@@ -269,60 +275,56 @@ void cleanup() {
      * completed in time.
      */
     // Visible for testing
-    void maybeAutoCommitAndLeaveGroup(final Timer timer) {
+    void maybeAutocommitOnClose(final Timer timer) {
         if (!requestManagers.coordinatorRequestManager.isPresent())
             return;
 
+        if (!requestManagers.commitRequestManager.isPresent()) {
+            log.error("Expecting a CommitRequestManager but the object was never initialized. Shutting down.");
+            return;
+        }
+
+        if (!requestManagers.commitRequestManager.get().canAutoCommit()) {
+            return;
+        }
+
         ensureCoordinatorReady(timer);
-        List<NetworkClientDelegate.UnsentRequest> tasks = closingRequests();
-        networkClientDelegate.addAll(tasks);
+        List<NetworkClientDelegate.UnsentRequest> autocommitRequest =
```

Comment:
Let's not use a list here, then we can drop the `get(0)` below. You can create the list in place in the line below or add a `NetworkClientDelegate.add` method.

Reply:
Cannot agree more!